src/eventhandling/click_handler.rs

```diff
@@ -0,0 +1,23 @@
+use amethyst::ecs::World;
+use amethyst::ecs::{Component, DenseVecStorage, Entity};
+use entities::buildings::Base;
+
+/// This trait is meant to be used as TraitObject to enable encapsulated implementation for every possible clickable Entity
+/// Note, that this does nothing, if the entity has no `HoverHandler` Component
+/// Note, that the for example `Base>>TraitObject` is an other Instance of `Base` than the Base Component of an entity itself. So if there is any state, that Clickable needs to access, the preferred way would be to add a dedicated Component to the entity, which implements this Trait
+///
+/// Note, that that `HoverHandler` is implemented as Component, which makes it harder to make Hoverable a supertrait of Clickable
+/// This can be adapted as soon as the is the need to do so @karyon <3<3<3
+pub trait Clickable: Sync + Send {
+    fn on_click(&self, &Entity, &World);
+}
+
+impl Component for Box<dyn Clickable> {
```

Comment:
The requirement is to have a `Storage<Clickable>`. You can do that in two ways:
**first:** see above
**second:** have a generic Clickable struct, that takes `Fn(entity:&Entity, data: SomeStorages)` and have different `Fn` for each clickable entity type.

However, you **cannot** have a `BaseClickable` and a `RRClickable` in the same Storage, since these are different Structs. They may have different sizes, have different members and implement different behavior. That's where `Box<dyn Clickable>` comes into play. Using a TraitObject (`dyn` ...) as Type allows us to store different Objects in the same Storage as long as they implement all `Clickable`. The `Box` is necessary, because a TraitObject's Size is not known at compile time which is a requirement for a Storage's type `T`. The combination `Box<dyn Clickable>` allows an encapsulated implementation of different types without having to collect from different storages.